src/applications/static-pages/cta-widget/index.js

```diff
@@ -133,8 +135,22 @@ export class CallToActionWidget extends Component {
     */
   }
 
+  updateReturnUrl = () => {
+    if (
+      this._internalLinkUrl.length > 0 &&
+      this._internalLinkUrl.startsWith('/')
+    ) {
+      // fix the internal link
+      sessionStorage.setItem(
+        'authReturnUrl',
+        `${window.location.origin}${this._internalLinkUrl}`,
+      );
+    }
+  };
+
```

Comment:
So one thing about the ctaWidgets is that each widget has a `redirect` boolean that can be returned from the `deriveToolUrlDetails` function.

So the updateReturnUrl should only be fired if `redirect` is set to true.

The `CTA_WIDGET_TYPES.DIRECT_DEPOSIT` widget settings does not have it set to redirect right now, but that is what we are hoping to set to true alongside this work.

Reply:
@adamwhitlock1 Can you elaborate more on the `redirect` property and its intended purpose?

The current solution for an unauthenticated user is that when they click on the **Sign in or create an account** they will be redirected to the URL specified in the `deriveToolUrlDetails` (this applies to all internal links not just Direct Deposit) 

The way I'm reading the above is that if they are already authenticated and the `redirect: true` it should **automatically** redirect them to the URL specified in the `deriveToolUrlDetails`?

Reply:
When `redirect: false` an unauthenticated user would see the sign in cta, would click the login button, and should then be returned back to the original url and not redirected to the url from `deriveToolUrlDetails`

Then upon returning to the original URL after sign in, they would see a cta with a button that would take them to the redirect url when clicked.

I'm not 100% sure on the intended purpose as far as _why_ a team doesn't want the redirect to be automatic, but I would guess that it may be a case when the cta lives on a teamsite like `http://www.benefits.va.gov/` since the automatic redirect wouldn't work as expected maybe?


Let me know if that makes sense. This is my first time dipping into this code as well, so my understanding may be off. It may be easier to get on a quick call if you have some time today or early next week.

Reply:
@adamwhitlock1 Okay I finally have some bandwidth to hop on a call to talk about this. Whenever you are ready :)

Reply:
@adamwhitlock1 Ready for review again. I will also ping my team for a review.